hantec_api_ecommerce/controllers/main.py

```diff
@@ -0,0 +1,799 @@
+        passed in the URL, and post the message in sale order
+
+        URL parameter:
+            - order (sale.order): The sale order model instance.
+
+        JSON request body:
+            - message_body (str): The content of the message to be posted.
+
+        JSON response:
+            - message (str): A confirmation message indicating the action performed.
+
+        Returns:
+            dict: A dictionary with a confirmation message.
+
+        """
+        message_body = request.jsonrequest.get("message_body")
+        # Post the message in the sale order chatter
+        order.message_post(body=message_body)
+        logger.debug("Message posted in sale order with ID %s", order.id)
+
+        return {
+            "message": f"Message successfully posted in sale order with ID: {order.id}."
+        }
+            
+    @route(["/get_inventory", "/get_inventory_by_sku"], methods=["POST", "GET"], type="json", auth="user")
+    def get_inventory(self):
+        """Retrieves the inventory details for all products with a SKU or a specific product based on its SKU from a given location.
+
+        This function searches for all products in the Odoo database that have a SKU (default_code not null)
+        or searches for a product using the provided SKU, and returns their inventory details from a given location
+        in a JSON response.
+
+        JSON request body for POST (if applicable):
+            - sku (str, optional): The SKU of the product to search for.
+            - location_id (int): The ID of the stock location to filter inventory by.
+
+        JSON response:
+            - message (str): A message indicating the action performed.
+            - inventory_data (list of dict): A list of dictionaries containing the inventory details for each product, including:
+                - name (str): The name of the product.
+                - default_code (str): The SKU of the product.
+                - qty_available (float): The quantity of the product available.
+                - virtual_available (float): The virtual quantity of the product available.
+
+        Returns:
+            dict: A dictionary with a message and the inventory details of the products.
+        """
+        location_id = request.jsonrequest.get("location_id") or request.params.get("location_id")
+        location = request.env["stock.location"].browse(int(location_id)).exists()
+        sku = request.jsonrequest.get("sku") if request.httprequest.method == 'POST' else request.params.get("sku")
+
+        if sku:
+            domain = [("default_code", "=", sku)]
+        else:
+            domain = [("default_code", "!=", False)]
+
+        products = request.env["product.product"].search(domain)
+
+        inventory_data = []
+        for product in products:
+            quants = request.env["stock.quant"].search([("product_id", "=", product.id), ("location_id", "=", location_id)])
```

Comment:
```suggestion
         product.with_context(location=location_id).read(["name", "default_code", "qty_available", "virtual_available"])
```
Don't need to look for quants